bolt-sidecar/src/primitives/commitment.rs

```diff
@@ -143,16 +144,67 @@ impl InclusionRequest {
         true
     }
 
-    pub fn validate_priority_fee(&self) -> bool {
+    /// Validates the priority fee against the max fee per gas.
+    /// Returns true if the fee is less than or equal to the max fee per gas, false otherwise.
+    pub fn validate_max_priority_fee(&self) -> bool {
         for tx in &self.txs {
-            if tx
-                .max_priority_fee_per_gas()
-                .is_some_and(|max_priority_fee| max_priority_fee > tx.max_fee_per_gas())
-            {
+            // Calculate the effective fee depending on the transaction type
+            let effective_fee = match tx.tx_type() {
+                TxType::Legacy | TxType::Eip2930 => {
+                    match tx.tx_type() {
+                        // Retrieve the gas price depending on the transaction type
+                        TxType::Legacy => tx.as_legacy().map(|tx| tx.gas_price),
+                        TxType::Eip2930 => tx.as_eip2930().map(|tx| tx.gas_price),
+                        _ => None,
+                    }
+                    .unwrap_or(0)
+                }
+                // For EIP-1559 (or any other), the effective priority fee is the max priority fee
+                // per gas
+                _ => tx.max_priority_fee_per_gas().unwrap_or(0),
+            };
```

Comment:
```suggestion
            let effective_fee = if tx.tx_type() == TxType::Legacy {
                tx.as_legacy().expect("legacy tx").gas_price
            } else if tx.tx_type() == TxType::Eip2930 {
                tx.as_eip2930().expect("Eip2930 tx").gas_price
            } else {
                // For EIP-1559 (or any other), the effective priority fee 
                // is the max priority fee per gas
                tx.max_priority_fee_per_gas().expect("Post EIP1559 tx")
            };

```
This is more a styling suggestion but lmk if you like it! I think it is more succint and having some good asserts can be helpful.